### ISSUE TITLE
[Feature Request] Support `min_p` Sampling Parameter

### ISSUE THREAD

**Issue by mcolala** [thu-pacman/chitu#122]
The current sampling implementation in `chitu/utils.py` includes placeholder code for `min_p` (minimum probability) sampling but it is not yet implemented. This is a commonly requested feature in LLM inference engines.

## Background

Looking at the codebase, there are multiple TODO comments indicating this feature is planned:

```python
# chitu/utils.py:511
# TODO: Support min_ps

# chitu/utils.py:542
# TODO: Support min_ps like: min_p_thresholds = probs_sort[:, 0] * min_ps

# chitu/utils.py:552
# TODO: Support min_ps like: probs_sort[probs_sort < min_p_thresholds.view(-1, 1)] = 0.0
```

## Proposed Implementation

`min_p` sampling filters out tokens with probability less than `min_p * max_probability`. The implementation would involve:

1. Add `min_ps` parameter to `top_k_top_p_min_p_sampling_from_logits()` function
2. Calculate threshold: `min_p_thresholds = probs_sort[:, 0] * min_ps`
3. Apply mask: `probs_sort[probs_sort < min_p_thresholds.view(-1, 1)] = 0.0`
4. Propagate parameter through `SampleParams`, `UserRequest`, and API endpoints

**Comment by roastduck** [comment]
Sorry for late rely. I am wondering in which model or applicaton would you need `min_p`? AFAIK, sampling parameters are less important than a year ago, as the capability of base models grow.
